--- WinQuake.org/vid_x.c	2021-12-03 22:11:00.635238952 +0200
+++ WinQuake/vid_x.c	2021-12-03 22:42:08.673346624 +0200
@@ -435,6 +435,25 @@
 
 }
 
+
+void
+setSizeHint(Display* display, Window window,
+            int minWidth, int minHeight,
+            int maxWidth, int maxHeight)
+{
+  XSizeHints hints = {};
+  if(minWidth > 0 && minHeight > 0) hints.flags |= PMinSize;
+  if(maxWidth > 0 && maxHeight > 0) hints.flags |= PMaxSize;
+
+  hints.min_width = minWidth;
+  hints.min_height = minHeight;
+  hints.max_width = maxWidth;
+  hints.max_height = maxHeight;
+
+  XSetWMNormalHints(display, window, &hints);
+}
+
+
 // Called at startup to set up translation tables, takes 256 8 bit RGB values
 // the palette data will go away after the call, so it must be copied off if
 // the video driver will need it again
@@ -495,6 +514,7 @@
 			Sys_Error("VID: -winsize <width> <height>\n");
 		vid.width = Q_atoi(com_argv[pnum+1]);
 		vid.height = Q_atoi(com_argv[pnum+2]);
+
 		if (!vid.width || !vid.height)
 			Sys_Error("VID: Bad window width/height\n");
 	}
@@ -589,6 +609,7 @@
 			x_vis,
 			attribmask,
 			&attribs );
+
 		XStoreName( x_disp,x_win,"xquake");
 
 
@@ -618,6 +639,7 @@
 		x_gc = XCreateGC(x_disp, x_win, valuemask, &xgcvalues );
 	}
 
+	setSizeHint(x_disp, x_win, 400, 300, 1920, 1024);
 // map the window
 	XMapWindow(x_disp, x_win);
 
